aya/src/sys/bpf.rs

```diff
@@ -46,6 +46,20 @@ pub(crate) fn bpf_create_map(
     u.map_type = def.map_type();
     u.key_size = def.key_size();
     u.value_size = def.value_size();
+    u.max_entries = def.max_entries();
+
+    // BPF_MAP_TYPE_PERF_EVENT_ARRAY's max_entries should be 0
+    // There are some (older) cases where this is a large constant
+    // Set to the number of online CPUs if it's 0
+    // Clamp to (at most) the number of online CPUs if it's larger than that
+    // Smaller values are left as is as this was required by some kernel selftests
```

Comment:
This comment is a bit difficult to read. I also don't understand two points:

- "There are some (older) cases where this is a large constant" - I didn't get it at all. Could you elaborate a bit?
- "Smaller values are left as is as this was required by some kernel selftests" - I don't get that as well. My understanding is that we should allow values greater than 0 and lower than number of CPUs for loading programs not written in Aya (or using old versions of aya-ebpf and hard coding high values). Is my understanding correct?

From what I was able to understand, I would rephrase it as:

```suggestion
    // BPF_MAP_TYPE_PERF_EVENT_ARRAY's max_entries should not exceed the number of
    // CPUs.
    //
    // By default, the newest versions of Aya, libbpf and cilium/ebpf define `max_entries` of
    // `PerfEventArray` as `0`, with an intention to get it replaced with a correct value
    // by the loader.
    //
    // We allow custom values (potentially coming either from older versions of aya-ebpf or
    // programs written in C) as long as they don't exceed the number of CPUs.
    //
    // Otherwise, when the value is `0` or too large, we set it to the number of CPUs.
```

Reply:
Good call. Incorporated your comment and re-pushed.